unicorn.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2017 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+module.exports = {
+  plugins: [
+    'unicorn',
+  ],
+  rules: {
+    // Improve regexes by making them shorter, consistent, and safer
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/better-regex.md
+    'unicorn/better-regex': 'error',
+
+    // Enforce passing a message value when throwing a built-in error
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/error-message.md
+    'unicorn/error-message': 'error',
+
+    // Require escape sequences to use uppercase values
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/escape-case.md
+    'unicorn/escape-case': 'error',
+
+    // Add expiration conditions to TODO comments
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/expiring-todo-comments.md
+    'unicorn/expiring-todo-comments': 'error',
+
+    // Enforce explicitly comparing the length property of a value
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/explicit-length-check.md
+    'unicorn/explicit-length-check': 'error',
+
+    // Enforce importing index files with .
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/import-index.md
+    'unicorn/import-index': 'error',
+
+    // Enforce the use of new for all builtins, except String, Number and Boolean
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/new-for-builtins.md
+    'unicorn/new-for-builtins': 'error',
+
+    // Enforce specifying rules to disable in eslint-disable comments
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-abusive-eslint-disable.md
+    'unicorn/no-abusive-eslint-disable': 'error',
+
+    // Require Array.isArray() instead of instanceof Array
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-array-instanceof.md
+    'unicorn/no-array-instanceof': 'error',
+
+    // Do not use leading/trailing space between console.log parameters
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-console-spaces.md
+    'unicorn/no-console-spaces': 'error',
+
+    // Do not use a for loop that can be replaced with a for-of loop
```

Comment:
why would we want this rule?

Reply:
its fixable and for-of loops can look cleaner

Reply:
are they as performant?

Reply:
Does not seem to be much data on it but looks like as usual "it depends", node version, total elements in array etc
https://medium.com/better-programming/which-is-the-fastest-while-for-foreach-for-of-9022902be15e